### PR TITLE
feat(remove-android): Docker file cleanup

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Details

Simplify our `Dockerfile` as follows:
- Remove the unified entry point
- Revert the Dockerfile changes from #5720, which is where the .NET build tools were added for mock-adb. Since the lines in this block were formatted inconsistently, adopt a consistent standard (one statement per line with && at the end of the line)

##### Motivation

Part of Android removal feature

#### Pull request checklist

- [x] Addresses an existing issue: Android deprecation feature
- [x] Ran `yarn fastpass`
- [n/a] Added/updated relevant unit test(s) (and ran `yarn test`)
- [n/a] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,17 +14,12 @@ USER root
 #
 # We pin nodejs 16.x instead of accepting Playwright's default for consistency with
 # our other build environments.
-#
-# dotnet-sdk-6.0 is required to build mock-adb
 RUN apt-get update && \
-  apt-get install -y curl apt-transport-https && \
```
Comment (DaveTryon) [inline Dockerfile:20]: This undoes the Dockerfile changes from #5720, then standardizes the formatting

File: Dockerfile [hunk 2]
```diff
@@ -14,17 +14,12 @@ USER root
 #
 # We pin nodejs 16.x instead of accepting Playwright's default for consistency with
 # our other build environments.
-#
-# dotnet-sdk-6.0 is required to build mock-adb
 RUN apt-get update && \
```
Comment (codeofdusk) [inline Dockerfile:17]: On modern Debian-based systems:

```suggestion
RUN apt update && \
```

File: Dockerfile [hunk 3]
```diff
@@ -14,17 +14,12 @@ USER root
...
-  apt-get install -y curl apt-transport-https && \
-  curl -fsSL https://deb.nodesource.com/setup_16.x | bash - && \
-  curl -fsSL https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -o packages-microsoft-prod.deb && \
-  dpkg -i packages-microsoft-prod.deb && \
-  rm packages-microsoft-prod.deb && \
+  apt-get install ca-certificates && \
```
Comment (codeofdusk) [inline Dockerfile:18]:
```suggestion
  apt install ca-certificates && \
```

File: Dockerfile [hunk 4]
```diff
@@ -14,17 +14,12 @@ USER root
...
   apt-get update && \
```
Comment (codeofdusk) [inline Dockerfile:19]:
```suggestion
  apt update && \
```

File: Dockerfile [hunk 5]
```diff
@@ -14,17 +14,12 @@ USER root
...
+  apt-get install -y curl && \
```
Comment (codeofdusk) [inline Dockerfile:20]:
```suggestion
  apt install -y curl && \
```

File: Dockerfile [hunk 6]
```diff
@@ -14,17 +14,12 @@ USER root
...
+  curl -fsSL https://deb.nodesource.com/setup_16.x | bash - && \
   apt-get install -y --allow-downgrades nodejs=16.* && \
```
Comment (codeofdusk) [inline Dockerfile:22]:
```suggestion
  apt install -y --allow-downgrades nodejs=16.* && \
```

**Review by codeofdusk — APPROVED** [review]

**Comment by DaveTryon** [comment]
@codeofdusk, I tried replacing `apg-get` with `apt` and the logs contained a warning that the apt CLI isn't stable and isn't recommended for scripting. Because of that, I reverted back to using `apt-get`

**Pull request merged**